<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#평가방식-MAE" data-toc-modified-id="평가방식-MAE-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>평가방식 MAE</a></span></li><li><span><a href="#library-import" data-toc-modified-id="library-import-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>library import</a></span></li><li><span><a href="#data-load" data-toc-modified-id="data-load-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>data load</a></span></li><li><span><a href="#전처리" data-toc-modified-id="전처리-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>전처리</a></span></li><li><span><a href="#optuna" data-toc-modified-id="optuna-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>optuna</a></span></li><li><span><a href="#model-fitting-with-tuned-hyperparameters" data-toc-modified-id="model-fitting-with-tuned-hyperparameters-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>model fitting with tuned hyperparameters</a></span></li></ul></div>

# 평가방식 MAE
... 회귀.... <br>
회귀도 input은 수치여야함.

# library import

In [1]:
import pandas as pd
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'
from tqdm import tqdm
import warnings
warnings.filterwarnings('ignore')

# data load

In [2]:
train = pd.read_csv('../235743_구내식당 식사 인원 예측 AI 경진대회_data/train.csv')
test = pd.read_csv('../235743_구내식당 식사 인원 예측 AI 경진대회_data/test.csv')

In [3]:
train.head(6)
# 일자
# 요일 ###
# 본사 정원수
# 본사휴가자수
# 본사출장자수
# 본사시간외근무명령서승인건수
# 현본사소속재택근무자수 ###
# 조식메뉴
# 중식메뉴 ###
# 석식메뉴 ###

# 중식계
# 석식계

# 중식계, 석식계는 합계인듯. 예측해야 하는 것.

일자 요일  본사정원수  본사휴가자수  본사출장자수  본사시간외근무명령서승인건수  현본사소속재택근무자수  \
0  2016-02-01  월   2601      50     150             238          0.0   
1  2016-02-02  화   2601      50     173             319          0.0   
2  2016-02-03  수   2601      56     180             111          0.0   
3  2016-02-04  목   2601     104     220             355          0.0   
4  2016-02-05  금   2601     278     181              34          0.0   
5  2016-02-11  목   2601     383     143             417          0.0   

                                                조식메뉴  \
0  모닝롤/찐빵  우유/두유/주스 계란후라이  호두죽/쌀밥 (쌀:국내산) 된장찌개  쥐...   
1  모닝롤/단호박샌드  우유/두유/주스 계란후라이  팥죽/쌀밥 (쌀:국내산) 호박젓국찌...   
2  모닝롤/베이글  우유/두유/주스 계란후라이  표고버섯죽/쌀밥 (쌀:국내산) 콩나물국...   
3  모닝롤/토마토샌드  우유/두유/주스 계란후라이  닭죽/쌀밥 (쌀,닭:국내산) 근대국...   
4  모닝롤/와플  우유/두유/주스 계란후라이  쇠고기죽/쌀밥 (쌀:국내산) 재첩국  방...   
5  팬케익/찐빵  우유/두유/주스  계란후라이  견과류죽/쌀밥 (쌀:국내산) 감자찌개 ...   

                                                중식메뉴  \
0  쌀밥/잡곡밥 (쌀,현미흑미:국내산) 오징어찌개  쇠불고기 (쇠고기:호주산) 계란찜 ...   
1  쌀밥/잡곡밥 (쌀,현미흑미:국내산) 김치찌개  가자미튀김  모둠소세지구이  마늘쫑무...   
2  카레덮밥 (쌀,현미흑미:국내산) 팽이장국  치킨핑거 (닭고기:국내산) 쫄면야채무침 ...   
3  쌀밥/잡곡밥 (쌀,현미흑미:국내산) 쇠고기무국  주꾸미볶음  부추전  시금치나물  ...   
4  쌀밥/잡곡밥 (쌀,현미흑미:국내산) 떡국  돈육씨앗강정 (돼지고기:국내산) 우엉잡채...   
5  쌀밥/잡곡밥 (쌀,현미흑미:국내산) 시래기국  훈제오리구이  도토리묵무침  쌈무/양...   

                                                석식메뉴     중식계    석식계  
0  쌀밥/잡곡밥 (쌀,현미흑미:국내산) 육개장  자반고등어구이  두부조림  건파래무침 ...  1039.0  331.0  
1  콩나물밥*양념장 (쌀,현미흑미:국내산) 어묵국  유산슬 (쇠고기:호주산) 아삭고추무...   867.0  560.0  
2  쌀밥/잡곡밥 (쌀,현미흑미:국내산) 청국장찌개  황태양념구이 (황태:러시아산) 고기...  1017.0  573.0  
3  미니김밥*겨자장 (쌀,현미흑미:국내산) 우동  멕시칸샐러드  군고구마  무피클  포...   978.0  525.0  
4  쌀밥/잡곡밥 (쌀,현미흑미:국내산) 차돌박이찌개 (쇠고기:호주산) 닭갈비 (닭고기:...   925.0  330.0  
5  참치회덮밥 (쌀,현미흑미:국내산) 맑은국  군만두  과일샐러드  락교  포기김치 (...  1045.0  550.0

# 전처리

In [4]:
# train['조식메뉴'].value_counts()
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1205 entries, 0 to 1204
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   일자              1205 non-null   object 
 1   요일              1205 non-null   object 
 2   본사정원수           1205 non-null   int64  
 3   본사휴가자수          1205 non-null   int64  
 4   본사출장자수          1205 non-null   int64  
 5   본사시간외근무명령서승인건수  1205 non-null   int64  
 6   현본사소속재택근무자수     1205 non-null   float64
 7   조식메뉴            1205 non-null   object 
 8   중식메뉴            1205 non-null   object 
 9   석식메뉴            1205 non-null   object 
 10  중식계             1205 non-null   float64
 11  석식계             1205 non-null   float64
dtypes: float64(3), int64(4), object(5)
memory usage: 113.1+ KB


In [5]:
train.loc[(train['조식메뉴'].str.contains('쇠'))|(train['조식메뉴'].str.contains('소'))|(train['조식메뉴'].str.contains('고기'))|
          (train['조식메뉴'].str.contains('돼지'))|(train['조식메뉴'].str.contains('돈육'))|
          (train['조식메뉴'].str.contains('닭'))|(train['조식메뉴'].str.contains('치킨'))|
          (train['조식메뉴'].str.contains('오리')), '조식메뉴'] = 1
train.loc[train['조식메뉴']!=1, '조식메뉴'] = 0
test.loc[(test['조식메뉴'].str.contains('쇠'))|(test['조식메뉴'].str.contains('소'))|(test['조식메뉴'].str.contains('고기'))|
          (test['조식메뉴'].str.contains('돼지'))|(test['조식메뉴'].str.contains('돈육'))|
          (test['조식메뉴'].str.contains('닭'))|(test['조식메뉴'].str.contains('치킨'))|
          (test['조식메뉴'].str.contains('오리')), '조식메뉴'] = 1
test.loc[test['조식메뉴']!=1, '조식메뉴'] = 0

# int 형으로 바꾸기
train['조식메뉴'] = train['조식메뉴'].map(int)
test['조식메뉴'] = test['조식메뉴'].map(int)

In [6]:
# train[train['중식메뉴'].contains('고기')]
# (train.loc[train['중식메뉴'].str.contains('고기'), '중식메뉴']).count()
train.shape
# train.loc[train['중식메뉴'].str.contains('고기'), '중식메뉴']
train.loc[(train['중식메뉴'].str.contains('쇠'))|(train['중식메뉴'].str.contains('소'))|(train['중식메뉴'].str.contains('고기'))|
          (train['중식메뉴'].str.contains('돼지'))|(train['중식메뉴'].str.contains('돈육'))|
          (train['중식메뉴'].str.contains('닭'))|(train['중식메뉴'].str.contains('치킨'))|
          (train['중식메뉴'].str.contains('오리')), '중식메뉴'] = 1
train.loc[train['중식메뉴']!=1, '중식메뉴'] = 0
# train.loc[(train['중식메뉴'].str.contains('돼지'))|(train['중식메뉴'].str.contains('돈육')), '중식메뉴']
# train.loc[(train['중식메뉴'].str.contains('닭'))|(train['중식메뉴'].str.contains('치킨')), '중식메뉴']
# train.loc[(train['중식메뉴'].str.contains('오리')), '중식메뉴']

test.loc[(test['중식메뉴'].str.contains('쇠'))|(test['중식메뉴'].str.contains('소'))|(test['중식메뉴'].str.contains('고기'))|
          (test['중식메뉴'].str.contains('돼지'))|(test['중식메뉴'].str.contains('돈육'))|
          (test['중식메뉴'].str.contains('닭'))|(test['중식메뉴'].str.contains('치킨'))|
          (test['중식메뉴'].str.contains('오리')), '중식메뉴'] = 1
test.loc[test['중식메뉴']!=1, '중식메뉴'] = 0

# int 형으로 바꾸기
train['중식메뉴'] = train['중식메뉴'].map(int)
test['중식메뉴'] = test['중식메뉴'].map(int)

(1205, 12)

In [7]:
train.loc[(train['석식메뉴'].str.contains('쇠'))|(train['석식메뉴'].str.contains('소'))|(train['석식메뉴'].str.contains('고기'))|
          (train['석식메뉴'].str.contains('돼지'))|(train['석식메뉴'].str.contains('돈육'))|
          (train['석식메뉴'].str.contains('닭'))|(train['석식메뉴'].str.contains('치킨'))|
          (train['석식메뉴'].str.contains('오리')), '석식메뉴'] = 1
train.loc[train['석식메뉴']!=1, '석식메뉴'] = 0

test.loc[(test['석식메뉴'].str.contains('쇠'))|(test['석식메뉴'].str.contains('소'))|(test['석식메뉴'].str.contains('고기'))|
          (test['석식메뉴'].str.contains('돼지'))|(test['석식메뉴'].str.contains('돈육'))|
          (test['석식메뉴'].str.contains('닭'))|(test['석식메뉴'].str.contains('치킨'))|
          (test['석식메뉴'].str.contains('오리')), '석식메뉴'] = 1
test.loc[test['석식메뉴']!=1, '석식메뉴'] = 0

# int 형으로 바꾸기
train['석식메뉴'] = train['석식메뉴'].map(int)
test['석식메뉴'] = test['석식메뉴'].map(int)

In [8]:
train['요일'].values

train['요일'] = train['요일'].map({'월':1, '화':2, '수':3, '목':4, '금':5})
test['요일'] = test['요일'].map({'월':1, '화':2, '수':3, '목':4, '금':5})

train['요일']

array(['월', '화', '수', ..., '금', '월', '화'], dtype=object)

0       1
1       2
2       3
3       4
4       5
       ..
1200    3
1201    4
1202    5
1203    1
1204    2
Name: 요일, Length: 1205, dtype: int64

In [9]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1205 entries, 0 to 1204
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   일자              1205 non-null   object 
 1   요일              1205 non-null   int64  
 2   본사정원수           1205 non-null   int64  
 3   본사휴가자수          1205 non-null   int64  
 4   본사출장자수          1205 non-null   int64  
 5   본사시간외근무명령서승인건수  1205 non-null   int64  
 6   현본사소속재택근무자수     1205 non-null   float64
 7   조식메뉴            1205 non-null   int64  
 8   중식메뉴            1205 non-null   int64  
 9   석식메뉴            1205 non-null   int64  
 10  중식계             1205 non-null   float64
 11  석식계             1205 non-null   float64
dtypes: float64(3), int64(8), object(1)
memory usage: 113.1+ KB


In [11]:
import lightgbm as lgbm
import random

# optuna
[공식 docs](https://optuna.readthedocs.io/_/downloads/en/latest/pdf/) <br>
[맨 처음 참고](https://ssoonidev.tistory.com/107) <br>
[dacon, 다수 알고리즘에 대한 optuna 코드 보유](https://dacon.io/codeshare/2876) <br>
[저장 안하려다 다른 포스팅 참고하라고 저장](https://velog.io/@dohy426/LGBM-Regressor-Optuna) <br>

In [12]:
lgbm.LGBMRegressor().get_params()

{'boosting_type': 'gbdt',
 'class_weight': None,
 'colsample_bytree': 1.0,
 'importance_type': 'split',
 'learning_rate': 0.1,
 'max_depth': -1,
 'min_child_samples': 20,
 'min_child_weight': 0.001,
 'min_split_gain': 0.0,
 'n_estimators': 100,
 'n_jobs': -1,
 'num_leaves': 31,
 'objective': None,
 'random_state': None,
 'reg_alpha': 0.0,
 'reg_lambda': 0.0,
 'silent': 'warn',
 'subsample': 1.0,
 'subsample_for_bin': 200000,
 'subsample_freq': 0}

In [25]:
# ! pip install optuna

In [14]:
import optuna
# from optuna.integration.lightgbm import LightGBMTuner as LGBMTuner
from optuna import Trial, visualization
from optuna.samplers import TPESampler
from sklearn.metrics import mean_absolute_error as MAE

def optuna_LGBM(trial: Trial, X_train_, y_train_, X_test_):
    parameters = {'objective': 'regression',
                  "random_state": 42,
                  'verbose': -1,
                  'metric': 'mae',
                  "silent": -1,
                  'learning_rate': trial.suggest_float('learning_rate', 1e-5, 1e-1, log = True),
                  'n_estimators': trial.suggest_int('n_estimators', 100, 5000),
                 'max_depth': trial.suggest_int('max_depth', 8, 16),
                 'min_child_weight': trial.suggest_int('min_child_weight', 1, 300),
                 'reg_alpha': trial.suggest_loguniform('reg_alpha', 1e-3, 10.0),
                 'reg_lambda': trial.suggest_loguniform('reg_lambda', 1e-3, 10.0)}

#     parameters = {
#         'objective': 'regression',
#         'verbose': -1,
#         'metric': 'mse', 
#         'num_leaves': trial.suggest_int('num_leaves', 2, 1024, step=1, log=True), 
#         'colsample_bytree': trial.suggest_uniform('colsample_bytree', 0.7, 1.0),
#         'reg_alpha': trial.suggest_uniform('reg_alpha', 0.0, 1.0),
#         'reg_lambda': trial.suggest_uniform('reg_lambda', 0.0, 10.0),
#         'max_depth': trial.suggest_int('max_depth',3, 15),
#         'learning_rate': trial.suggest_loguniform("learning_rate", 1e-8, 1e-2),
#         'n_estimators': trial.suggest_int('n_estimators', 100, 3000),
#         'min_child_samples': trial.suggest_int('min_child_samples', 5, 100),
#         'subsample': trial.suggest_loguniform('subsample', 0.4, 1),
#     }
    user_seed = 42
    random.seed(user_seed)
#     LGBM = lgbm.LGBMRegressor() # 이런식으로 하면 자꾸 에러남.

    optuna_model = lgbm.LGBMRegressor(**parameters)
    lgb_model = optuna_model.fit(X_train_, y_train_)
    score = MAE(y_train_, lgb_model.predict(X_train_))
    
    return score


X_train_중식 = train[['요일', '현본사소속재택근무자수', '중식메뉴']]
X_train_석식 = train[['요일', '현본사소속재택근무자수', '석식메뉴']]

y_train_중식 = train[['중식계']]
y_train_석식 = train[['석식계']]

X_test_중식 = test[['요일', '현본사소속재택근무자수', '중식메뉴']]
X_test_석식 = test[['요일', '현본사소속재택근무자수', '석식메뉴']]


study_중식 = optuna.create_study(direction = 'minimize', sampler = TPESampler(seed = 42))
study_석식 = optuna.create_study(direction = 'minimize', sampler = TPESampler(seed = 42))


study_중식.optimize(lambda trial : optuna_LGBM(trial, X_train_중식, y_train_중식, X_test_중식), n_trials = 50)
best_parameters_중식 = study_중식.best_trial.params
print(f'중식 Best trial: score {study_중식.best_trial.value}, \nparams {best_parameters_중식}')

study_석식.optimize(lambda trial : optuna_LGBM(trial, X_train_석식, y_train_석식, X_test_석식), n_trials = 50)
best_parameters_석식 = study_석식.best_trial.params
print(f'석식 Best trial: score {study_석식.best_trial.value}, \nparams {best_parameters_석식}')

# [scoring method name](https://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter)

[I 2022-07-06 01:28:28,143] A new study created in memory with name: no-name-9201e154-b87b-489a-97db-6242059b5620
[I 2022-07-06 01:28:28,144] A new study created in memory with name: no-name-1d67ea3b-91bd-42de-9db0-968f27316d73
[I 2022-07-06 01:28:31,545] Trial 0 finished with value: 106.26033547194069 and parameters: {'learning_rate': 0.00031489116479568613, 'n_estimators': 4759, 'max_depth': 14, 'min_child_weight': 180, 'reg_alpha': 0.004207988669606638, 'reg_lambda': 0.004207053950287938}. Best is trial 0 with value: 106.26033547194069.
[I 2022-07-06 01:28:34,693] Trial 1 finished with value: 161.05804331124884 and parameters: {'learning_rate': 1.7073967431528103e-05, 'n_estimators': 4345, 'max_depth': 13, 'min_child_weight': 213, 'reg_alpha': 0.0012087541473056963, 'reg_lambda': 7.579479953348009}. Best is trial 0 with value: 106.26033547194069.
[I 2022-07-06 01:28:36,362] Trial 2 finished with value: 92.09418111155519 and parameters: {'learning_rate': 0.021368329072358756, 'n_esti

[I 2022-07-06 01:29:55,795] Trial 25 finished with value: 92.41481227070834 and parameters: {'learning_rate': 0.007746245538301296, 'n_estimators': 2786, 'max_depth': 10, 'min_child_weight': 69, 'reg_alpha': 0.04136899942732345, 'reg_lambda': 0.06807412558779614}. Best is trial 11 with value: 86.22319136792808.
[I 2022-07-06 01:29:56,013] Trial 26 finished with value: 95.29357611530146 and parameters: {'learning_rate': 0.018308588617871217, 'n_estimators': 156, 'max_depth': 12, 'min_child_weight': 47, 'reg_alpha': 0.14312833192746546, 'reg_lambda': 0.2286862657227274}. Best is trial 11 with value: 86.22319136792808.
[I 2022-07-06 01:29:58,776] Trial 27 finished with value: 87.96918358067396 and parameters: {'learning_rate': 0.052598230323601745, 'n_estimators': 1445, 'max_depth': 11, 'min_child_weight': 1, 'reg_alpha': 0.8653055264440151, 'reg_lambda': 0.014089463646131955}. Best is trial 11 with value: 86.22319136792808.
[I 2022-07-06 01:30:00,369] Trial 28 finished with value: 127.77

Best trial: score 86.22319136792808, 
params {'learning_rate': 0.08908472316730853, 'n_estimators': 2477, 'max_depth': 11, 'min_child_weight': 2, 'reg_alpha': 0.38226021752753925, 'reg_lambda': 0.6146439247862426}


[I 2022-07-06 01:31:15,592] Trial 0 finished with value: 85.04259711812013 and parameters: {'learning_rate': 0.00031489116479568613, 'n_estimators': 4759, 'max_depth': 14, 'min_child_weight': 180, 'reg_alpha': 0.004207988669606638, 'reg_lambda': 0.004207053950287938}. Best is trial 0 with value: 85.04259711812013.
[I 2022-07-06 01:31:18,337] Trial 1 finished with value: 97.82490614145108 and parameters: {'learning_rate': 1.7073967431528103e-05, 'n_estimators': 4345, 'max_depth': 13, 'min_child_weight': 213, 'reg_alpha': 0.0012087541473056963, 'reg_lambda': 7.579479953348009}. Best is trial 0 with value: 85.04259711812013.
[I 2022-07-06 01:31:19,828] Trial 2 finished with value: 76.48226178730314 and parameters: {'learning_rate': 0.021368329072358756, 'n_estimators': 1140, 'max_depth': 9, 'min_child_weight': 56, 'reg_alpha': 0.016480446427978974, 'reg_lambda': 0.12561043700013558}. Best is trial 2 with value: 76.48226178730314.
[I 2022-07-06 01:31:21,948] Trial 3 finished with value: 84

[I 2022-07-06 01:32:27,456] Trial 26 finished with value: 79.77226580202606 and parameters: {'learning_rate': 0.018308588617871217, 'n_estimators': 156, 'max_depth': 12, 'min_child_weight': 47, 'reg_alpha': 0.14312833192746546, 'reg_lambda': 0.2286862657227274}. Best is trial 11 with value: 71.4076828159998.
[I 2022-07-06 01:32:29,624] Trial 27 finished with value: 73.39658552286414 and parameters: {'learning_rate': 0.052598230323601745, 'n_estimators': 1445, 'max_depth': 11, 'min_child_weight': 1, 'reg_alpha': 0.8653055264440151, 'reg_lambda': 0.014089463646131955}. Best is trial 11 with value: 71.4076828159998.
[I 2022-07-06 01:32:30,974] Trial 28 finished with value: 89.99631315043085 and parameters: {'learning_rate': 0.004316771866635478, 'n_estimators': 3618, 'max_depth': 14, 'min_child_weight': 297, 'reg_alpha': 0.37914903969895764, 'reg_lambda': 0.5146530724904312}. Best is trial 11 with value: 71.4076828159998.
[I 2022-07-06 01:32:34,500] Trial 29 finished with value: 76.465377

Best trial: score 71.4076828159998, 
params {'learning_rate': 0.08908472316730853, 'n_estimators': 2477, 'max_depth': 11, 'min_child_weight': 2, 'reg_alpha': 0.38226021752753925, 'reg_lambda': 0.6146439247862426}


# model fitting with tuned hyperparameters
중식 Best trial: score 86.22319136792808, <br>
params {'learning_rate': 0.08908472316730853, 'n_estimators': 2477, 'max_depth': 11, 'min_child_weight': 2, 'reg_alpha': 0.38226021752753925, 'reg_lambda': 0.6146439247862426} <br>

석식 Best trial: score 71.4076828159998, <br>
params {'learning_rate': 0.08908472316730853, 'n_estimators': 2477, 'max_depth': 11, 'min_child_weight': 2, 'reg_alpha': 0.38226021752753925, 'reg_lambda': 0.6146439247862426} <br>

In [17]:
X_train_중식 = train[['요일', '현본사소속재택근무자수', '중식메뉴']]
X_train_석식 = train[['요일', '현본사소속재택근무자수', '석식메뉴']]
y_train_중식 = train[['중식계']]
y_train_석식 = train[['석식계']]

X_test_중식 = test[['요일', '현본사소속재택근무자수', '중식메뉴']]
X_test_석식 = test[['요일', '현본사소속재택근무자수', '석식메뉴']]

user_seed = 42
random.seed(user_seed)
LGBM = lgbm.LGBMRegressor(random_state = user_seed, learning_rate= 0.08908472316730853, n_estimators= 2477, max_depth= 11, min_child_weight= 2, reg_alpha= 0.38226021752753925, reg_lambda= 0.6146439247862426)

model_중식 = LGBM.fit(X_train_중식, y_train_중식) # fit에는 하이퍼파리미터 따로 적용 못하나봄. 계속 여기다 hp 집어넣을 때마다 에러남.
y_pred_test_중식 = model_중식.predict(X_test_중식, num_iteration = LGBM.best_iteration_)

model_석식 = LGBM.fit(X_train_석식, y_train_석식)
y_pred_test_석식 = model_석식.predict(X_test_석식, num_iteration = LGBM.best_iteration_)


In [18]:
중식계 = pd.DataFrame(y_pred_test_중식)
중식계.head()

0
0  1115.206060
1   892.774826
2   736.846166
3  1136.199742
4   913.609199

In [19]:
석식계 = pd.DataFrame(y_pred_test_석식)
석식계.head()

0
0  483.365236
1  460.136951
2  293.099286
3  530.538064
4  554.303601

In [20]:
findata = pd.concat([test['일자'], 중식계, 석식계], axis = 1)
findata.head()

일자            0           0
0  2021-01-27  1115.206060  483.365236
1  2021-01-28   892.774826  460.136951
2  2021-01-29   736.846166  293.099286
3  2021-02-01  1136.199742  530.538064
4  2021-02-02   913.609199  554.303601

In [21]:
findata.columns = ['일자', '중식계', '석식계']
findata.head()

일자          중식계         석식계
0  2021-01-27  1115.206060  483.365236
1  2021-01-28   892.774826  460.136951
2  2021-01-29   736.846166  293.099286
3  2021-02-01  1136.199742  530.538064
4  2021-02-02   913.609199  554.303601

In [22]:
findata.to_csv('7th optuna 요일, 재택자수, 중식메뉴, 석식메뉴.csv', index=False)